app/models/repository.rb

```diff
@@ -80,8 +80,14 @@ def self.search(
 
     matched_by_user = readable_rows.joins(:created_by).where_attributes_like('users.full_name', query, options)
 
+    prefixed_repository_row_ids = RepositoryRow.where(repository_id: repositories)
+                                               .where_attributes_like([RepositoryRow::PREFIXED_ID_SQL], query, options)
+                                               .pluck(:id)
```

Comment:
What if you have million rows and search string is 'IT1'?

Reply:
Previously, this was doing a search by "id LIKE ..." which I think was even worse. Should we just remove this from global search? Although I don't think it would ever reach millions of records.

Reply:
and now you simply insert ids as query parameter?

Reply:
fixed to use subquery